src/Types/ListType.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace ILIAS\Types;
+
+use ILIAS\Types\Exceptions\InvalidArgumentException;
+
+class ListType implements Type {
+
+	/** @var Type */
+	protected $type;
+
+	/**
+	 * SingleType constructor.
+	 * @param $fullyQualifiedClassName string|Type Give a Value Type or a Type that will be wrapped in a single type.
+	 */
+	public function __construct($fullyQualifiedClassName) {
+		if(!is_a($fullyQualifiedClassName, Type::class))
+			$fullyQualifiedClassName = new SingleType($fullyQualifiedClassName);
```

Comment:
I like reducing clutter, but would support @Amstutz on this one. Having brackets after every `if` helps reducing bugs when adding additional statements to the block of the `if`. At least common dev folklore says so...